src/main/java/net/snowflake/client/core/arrow/MapConverter.java

```diff
@@ -21,13 +28,20 @@ public MapConverter(MapVector valueVector, int columnIndex, DataConversionContex
   public Object toObject(int index) throws SFException {
     List<JsonStringHashMap<String, Object>> entriesList =
         (List<JsonStringHashMap<String, Object>>) vector.getObject(index);
-    return entriesList.stream()
-        .collect(
-            Collectors.toMap(entry -> entry.get("key").toString(), entry -> entry.get("value")));
+    return StructuredTypeConversionHelper.mapHashMapToObject(entriesList);
   }
 
   @Override
   public String toString(int index) throws SFException {
-    return vector.getObject(index).toString();
+    try {
+      return objectMapper.writeValueAsString(toObject(index));
+    } catch (JsonProcessingException e) {
+      throw new SFException(ErrorCode.INVALID_VALUE_CONVERT, e.getMessage());
+    }
+  }
+
+  @Override
+  public byte[] toBytes(int index) throws SFException {
+    return toString(index).getBytes();
```

Comment:
Without encoding? Can we have a test with `zażółć gęślą jaźń`? ;) Add the test before, I'm curious about test result anyway!

Reply:
Added test case